common/test/java/com/couchbase/lite/LiveQueryTest.java

```diff
@@ -51,108 +54,107 @@ public void testCreateBasicListener() throws InterruptedException {
             .orderBy(Ordering.property(KEY).ascending());
 
         final CountDownLatch latch = new CountDownLatch(1);
-
         ListenerToken token = query.addChangeListener(testSerialExecutor, change -> latch.countDown());
-
-        try { assertTrue(latch.await(TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS)); }
+        try { assertTrue(latch.await(TOLERABLE_QUERY_RUN_TIME, TimeUnit.MILLISECONDS)); }
         finally { query.removeChangeListener(token); }
     }
 
-    // Test create a second query, first query shouldn't get call back
-    // Test flaky with 220ms
-    // !!! This test does not test what it claims to test
-    @Ignore("Test flaky with 220ms")
+    /**
+     * When a second observer is registered, it should get call back after query done running
+     * The first observer should NOT get notified when the second observer is created
+     * When there's a db change, both observers should get notified in a tolerable amount of time
+     */
     @Test
-    public void testCreateSecondListener() throws InterruptedException, CouchbaseLiteException {
+    public void testMultipleListeners() throws InterruptedException, CouchbaseLiteException {
         final Query query = QueryBuilder
             .select(SelectResult.expression(Meta.id))
             .from(DataSource.database(baseTestDb))
             .where(Expression.property(KEY).greaterThanOrEqualTo(Expression.intValue(0)))
             .orderBy(Ordering.property(KEY).ascending());
 
-        final CountDownLatch latch1 = new CountDownLatch(2);
+        final CountDownLatch[] latch1 = new CountDownLatch[2];
         final CountDownLatch[] latch2 = new CountDownLatch[2];
-        for (int i = 0; i < latch2.length; i++) { latch2[i] = new CountDownLatch(1);}
-        final int[] count = new int[] {0};
 
-        // latch 0 count down once when adding listener
-        ListenerToken token1 = query.addChangeListener(testSerialExecutor, change -> latch1.countDown());
-        ListenerToken token2 = query.addChangeListener(testSerialExecutor, change -> {
-            int n = count[0]++;
-            latch2[n].countDown();
-        });
+        for (int i = 0; i < latch1.length; i++) { latch1[i] = new CountDownLatch(1); }
+        for (int i = 0; i < latch2.length; i++) { latch2[i] = new CountDownLatch(1); }
+
+        final int[] count = new int[] {0, 0};
+        ListenerToken token1 = query.addChangeListener(
+            testSerialExecutor,
+            change -> {
+                int n = count[0]++;
+                latch1[n].countDown();
+            });
+
+        // listener 1 gets notified after observer subscribed
+        assertTrue(latch1[0].await(TOLERABLE_QUERY_RUN_TIME, TimeUnit.MILLISECONDS));
```

Comment:
If this assert fails the test will terminate without removing the first change listener.